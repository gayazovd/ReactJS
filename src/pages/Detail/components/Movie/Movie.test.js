import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import FilmDetailView from './view';

configure({ adapter: new Adapter() });

describe('Component is render', () => {
  const props = {
    data: {
      title: 'Heroes',
      tagline: 'movie',
      vote_average: 7.7,
      release_date: '2019',
      poster_path: 'path',
      overview: 'asas',
      runtime: '120'
    }
  };
  const filmView = shallow(<FilmDetailView {...props} />);

  it('<FilmDetail /> snapshot', () => {
    expect(filmView).toMatchSnapshot();
  });

  it('detail component is correct', () => {
    expect(filmView.find('.title').text()).toEqual(props.data.title);
  });
});
