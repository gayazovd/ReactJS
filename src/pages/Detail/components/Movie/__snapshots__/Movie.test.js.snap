// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component is render <FilmDetail /> snapshot 1`] = `
<div
  className="wrapper"
>
  <div
    className="buttonsWrapper"
  >
    <p
      className="logo"
    >
      netflixroulette
    </p>
    <button
      className="search"
    >
      SEARCH
    </button>
  </div>
  <div
    className="film-present"
  >
    <img
      className="film-image"
      src="path"
    />
    <div
      className="description"
    >
      <div
        className="wrapper-title-average"
      >
        <h1
          className="title"
        >
          Heroes
        </h1>
        <div
          className="vote-average"
        >
          7.7
        </div>
      </div>
      <h5>
        movie
      </h5>
      <div
        className="release-date"
      >
        <span>
          2019
        </span>
        <span>
          120
           min
        </span>
      </div>
      <p
        className="overview"
      >
        asas
      </p>
    </div>
  </div>
</div>
`;
